sql_generators/glean_usage/common.py

```diff
@@ -223,17 +237,31 @@ def generate_per_app_id(
             return
 
         if output_dir:
-            write_sql(
-                output_dir, view, "metadata.yaml", view_metadata, skip_existing=True
-            )
-            write_sql(output_dir, view, "view.sql", view_sql, skip_existing=True)
-            write_sql(
-                output_dir, table, "metadata.yaml", table_metadata, skip_existing=True
-            )
-            write_sql(output_dir, table, "query.sql", query_sql, skip_existing=True)
+            # generated files to update
+            Artifact = namedtuple("Artifact", "table_id basename sql")
+            artifacts = [
+                Artifact(view, "metadata.yaml", view_metadata),
+                Artifact(view, "view.sql", view_sql),
+                Artifact(table, "metadata.yaml", table_metadata),
+                Artifact(table, "query.sql", query_sql),
+            ]
 
             if not self.no_init:
-                write_sql(output_dir, table, "init.sql", init_sql, skip_existing=True)
+                artifacts.append(Artifact(table, "init.sql", init_sql))
```

Comment:
This keeps the existing behavior of assuming an `init.sql` file, but per the logic above that sets `init_sql` the query file may use the `is_init()` macro rather than having a separate `init.sql` file.

Reply:
yeah, I think this could use some additional consideration. For now I think keeping the same behaviour and opening a follow-up ticket is the way to go